storage/src/mmr/proof.rs

```diff
@@ -1116,20 +1181,22 @@ mod tests {
             proof.verify_range_inclusion_and_extract_digests(
                 &mut hasher,
                 &elements,
-                0,
+                Location::new(0),
                 &wrong_root
             ),
             Err(Error::RootMismatch)
         ));
 
         // Test 2: Single element range (first element)
         let range = 0..1;
```

Comment:
I would define this range either as a range of locations or a range of usize, so you can avoid some of the conversions below.  (Same comment for other ranges in this test case)

Reply:
Done. Unfortunately the orphan rule prevents us from implementing `From<Range<Location>> for Range<usize>` and vice versa, which would make the code cleaner :(